bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -446,19 +437,25 @@ void PreProcessor::releaseClientPreProcessRequestSafe(uint16_t clientId) {
   releaseClientPreProcessRequest(clientEntry, clientId);
 }
 
+// This function should be always called under a clientEntry->mutex lock
 void PreProcessor::releaseClientPreProcessRequest(const ClientRequestStateSharedPtr &clientEntry, uint16_t clientId) {
-  if (clientEntry->reqProcessingStatePtr) {
-    LOG_DEBUG(logger(),
-              "clientId=" << clientId << " requestSeqNum=" << clientEntry->reqProcessingStatePtr->getReqSeqNum()
-                          << " released");
-    clientEntry->reqProcessingStatePtr.reset();
+  auto &givenReq = clientEntry->reqProcessingStatePtr;
+  if (givenReq) {
```

Comment:
when would this be `false` ?

Reply:
It could in case it was released in between.